app/javascript/packages/i18n/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @typedef {typeof globalThis & {
+ *   _locale_data?: Record<string, string>
+ * }} GlobalWithLocaleData
+ */
+
+/**
+ * Returns string with variable substitution.
+ *
+ * @param {string} string Original string.
+ * @param {Record<string,string>} variables Variables to replace.
+ *
+ * @return {string} String with variables substituted.
+ */
+export function replaceVariables(string, variables) {
+  return Object.keys(variables).reduce(
+    (result, key) => result.replace(new RegExp(`%{${key}}`, 'g'), variables[key]),
```

Comment:
I know this just moved from another file, but seeing the Regexp construction makes me wish JS had a method like Ruby's [Regexp.escape](https://ruby-doc.org/core-3.0.1/Regexp.html#method-c-escape)

Reply:
> I know this just moved from another file, but seeing the Regexp construction makes me wish JS had a method like Ruby's [Regexp.escape](https://ruby-doc.org/core-3.0.1/Regexp.html#method-c-escape)

Yeah, it's a good call-out, especially as we try to make this more _robust_. The standards body is/has [considered such a feature](https://github.com/tc39/proposal-regex-escaping), which is available in [a polyfill](https://www.npmjs.com/package/regexp.escape). There's also a [version in Lodash](https://lodash.com/docs/4.17.15#escapeRegExp), which we've avoided 'til now, but could also tie to your other comment about `snakeCase`, since it'd be nice to just use [an existing utility](https://lodash.com/docs/4.17.15#snakeCase). I had even implemented an escape function in an earlier iteration of the [library I mentioned earlier](https://github.com/aduth/terpolate), though I changed the implementation and can't recall it. Wouldn't be too hard to reimplement though, using [these characters](https://github.com/tc39/proposal-regex-escaping/blob/main/EscapedChars.md) as reference.

Reply:
I solved this a different way, by inverting the logic: Rather than create patterns for each key of the variables object, we create a single pattern which matches any possible key, and use the variable value.

See: 2156dd9

Reply:
love it 👍